lib/jekyll/commands/serve.rb

```diff
@@ -33,6 +33,8 @@ def init_with_program(prog)
               opts["serving"] = true
               opts["watch"  ] = true unless opts.key?("watch")
               config = opts["config"]
+              url = default_url(opts)
+              opts = opts.merge({ "url" => url }) if Jekyll.env == "development"
```

Comment:
Ah thanks, i only tried it with the non-bang version and then the line was too long :)